client/src/components/CreatePost.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { useState } from "react";
+import CssBaseline from "@mui/material/CssBaseline";
+
+import { createTheme, ThemeProvider } from "@mui/material/styles";
+import Grid from "@mui/material/Grid";
+import Box from "@mui/material/Box";
+import Container from "@mui/material/Container";
+import Button from "@mui/material/Button";
+import TextField from "@mui/material/TextField";
+import Divider from "@mui/material/Divider";
+import Typography from "@mui/material/Typography";
+
+import Dialog from "@mui/material/Dialog";
+import DialogContent from "@mui/material/DialogContent";
+import DialogActions from "@mui/material/DialogActions";
+
+import PreviewPopUp from "./PreviewPopUp";
+import { DialogTitle } from "@mui/material";
+
+const theme = createTheme();
+
+function CreatePost() {
+  // create hooks for input fields
+  const [title, setTitle] = useState("");
+  const [body, setBody] = useState("");
+  const [file, setFile] = useState("");
+  const [tags, setTags] = useState("");
+  const [eventCapacity, setEventCapacity] = useState("");
+  const [eventLocation, setEventLocation] = useState("");
```

Comment:
I believe @danielduong8 is storing form state in an object, lets stay consistent across pages. I'm fine with either approach.

Reply:
Can you explain what you mean by storing form state in an object? I'm not familiar

Reply:
e.g.
```
const [form, setForm] = useState({
    title: "",
    body: "",
    file: "",
    tags: "",
    eventCapacity: "",
    ....
  });
```

Reply:
To set things you would have to use something like:
```
setForm({ ...form, [e.target.name]: e.target.value });
```
Keep in mind you would also have to set the `name=blah` properties on the components where you use this handler.

Reply:
If you're not comfortable with doing it, you can leave this for now and I can take care of it once I issue the PR for connecting all the backend to the frontend.

Reply:
okay that would be good, but I think I get it though

Reply:
Still TODO?